modules/swagger-codegen/src/main/resources/asyncscala/api.mustache

```diff
@@ -40,9 +40,14 @@ class {{classname}}(client: TransportClient, config: SwaggerConfig) extends ApiC
     {{/required}}
     {{/queryParams}}
 
-    {{#headerParams}}
-    headerParams += "{{baseName}}" -> {{paramName}}.toString
-    {{/headerParams}}
+  {{#headerParams}}
+      {{^required}}
+          if ({{paramName}} != null) {{paramName}}.map { v => headerParams += "{{baseName}}" -> v.toString }
+      {{/required}}
+      {{#required}}
+          if ({{paramName}} != null) headerParams += "{{baseName}}" -> {{paramName}}.toString
+      {{/required}}
```

Comment:
Yes, basically we will never see null in Scala code. And actually that's the reason why we have Option.